syntax = "proto3";

package AUTH_RPC;

service AuthService {
  // 图形验证码
  rpc GetCaptchaImage (CaptchaImageRequest) returns (CaptchaImageResponse); 
  rpc VerifyCaptchaImage (CaptchaImageVerification) returns (Status);

  // 邮箱验证码
  rpc GetCaptchaEmail (CaptchaEmailRequest) returns (CaptchaImageResponse); 
  rpc VerifyCaptchaEmail (CaptchaEmailVerification) returns (Status);
}

///////////////////////////////////////////////////////////////////
// 图形验证码
///////////////////////////////////////////////////////////////////

// 前端获取图形验证码请求
message CaptchaImageRequest {}

// 返回前端图形验证码
message CaptchaImageResponse {
  string captcha_image_id = 1;
  string captcha_image_base64 = 2;
}

// 后端校验图形验证码
message CaptchaImageVerification {
  string captcha_image_id = 1;
  string captcha_image_value = 2;
}


///////////////////////////////////////////////////////////////////
// 邮箱验证码
///////////////////////////////////////////////////////////////////

// 前端获取邮箱验证码请求
message CaptchaEmailRequest {
  string email = 1;
}

// 返回前端邮箱验证码
message CaptchaEmailResponse {
  string captcha_email_id = 1;
}

// 后端校验邮箱验证码
message CaptchaEmailVerification {
  string captcha_email_id = 1;
  string captcha_email_value = 2;
}


///////////////////////////////////////////////////////////////////
// 登录注册
///////////////////////////////////////////////////////////////////


///////////////////////////////////////////////////////////////////
// JWT令牌
///////////////////////////////////////////////////////////////////


///////////////////////////////////////////////////////////////////
// 阿里云OSS对象管理 STS_TOKEN 临时授权
///////////////////////////////////////////////////////////////////


// 验证码校验状态
message Status {
    bool status = 1;
}